bokeh/charts/builder.py

```diff
@@ -114,24 +113,31 @@ class Builder(HasProps):
     yscale = String()
 
     # Dimension Configuration
-    dimensions = List(String, help="""The dimension
-        labels that drive the position of the glyphs. Subclasses should implement this
-        so that the Builder base class knows which dimensions it needs to operate on.
-        An example for a builder working with cartesian x and y coordinates would be
-        dimensions = ['x', 'y']. You should then instantiate the x and y dimensions as
-        attributes of the subclass of builder using the
-        :class:`Dimension <bokeh.charts.properties.Dimension>` class. One for x,
-        as x = Dimension(...), and one as y = Dimension(...).
-        """)
 
-    req_dimensions = Either(List(String), List(List(String)), List(Dict(String, String)),
-                            help="""
-        The dimension labels that must exist to produce the glyphs. This specifies what
-        are the valid configurations for the chart, with the option of specifying the
-        type of the columns. The :class:`~bokeh.charts.data_source.ChartDataSource` will
-        inspect this property of your subclass of Builder and use this to fill in any
-        required dimensions if no keyword arguments are used.
-        """)
+    """
+    The dimension labels that drive the position of the
+    glyphs. Subclasses should implement this so that the Builder
+    base class knows which dimensions it needs to operate on.
+    An example for a builder working with cartesian x and y
+    coordinates would be dimensions = ['x', 'y']. You should
+    then instantiate the x and y dimensions as attributes of the
+    subclass of builder using the :class:`Dimension
+    <bokeh.charts.properties.Dimension>` class. One for x, as x
+    = Dimension(...), and one as y = Dimension(...).
+    """
```

Comment:
The original intent with these properties that are overridden was to utilize the type checking, but I was wanting to avoid having each subclass implement __init__ and pass in the subclass-specific values that way, but didn't see a way to do this with the properties system. Since this wasn't an immediate priority, I left it as is, since it provided the documentation and modeled what the attributes should look like. I have a todo to go back and figure out a better way to do what I was trying to do. Since hasprops models instance properties, it was a misuse anyways.